data/store/mongo/mongo_datum.go

```diff
@@ -543,6 +545,30 @@ func validateAndTranslateSelectors(selectors *data.Selectors) (bson.M, error) {
 	return selector, nil
 }
 
+func (d *DatumRepository) CheckDataSetContainsType(ctx context.Context, dataSetID string, typ string) (bool, error) {
+	twoYearsPast := time.Now().UTC().AddDate(0, -24, 0)
+	oneDayFuture := time.Now().UTC().AddDate(0, 0, 1)
+
+	selector := bson.M{
+		"_active":  true,
+		"uploadId": dataSetID,
+		"type":     typ,
+		"time": bson.M{"$gt": twoYearsPast,
```

Comment:
This query doesn't do exactly what the function name says it does - consider passing the time bounds as a parameter